test/logs.spec.tsx

```diff
@@ -36,7 +36,7 @@ describe("Components with logs ", () => {
     fireEvent.mouseEnter(loader);
     const arcConnectionError = await findByText(
       container,
-      /No connection to the graph/
+      /this.props.config.initialize is not a function/
```

Comment:
This is a javascript runtime error, and not a "connection error"... the reason this is happening is because on line 25 you're incorrectly spreading (`...`) then casting a class like so `{ ...arcConfig, graphqlHttpUrl: "wrong url" } as ArcConfig`. When you spread (`...`) a class instance (`arcConfig`), it turns strips away all **methods** on the object. This is because it removes the `this` pointer from the object instance, and the **methods** rely on being bound to a `this` pointer.

Please consider this code sample:
```typescript
class A {
  method() {
    // this.somethingUseful
  }

  prop: number
  func: () => void
}

const a = new A()
const b = {...a}

b.prop // exists
b.func // exists, as this is a normal property on the class
b.method // doesn't exist! No more this pointer for the method to be bound to
```

A good rule of thumb is that whenever you have to cast something, you should think a bit more about why that is and try to avoid it at all costs because chances are you're doing something wrong.

A way to do this properly is like so:
```typescript
const badConfig = new ArcConfig(
      arcConfig.web3HttpUrl,
      "wrong url",
      arcConfig.graphqlWsUrl,
      arcConfig.ipfsProvider,
      arcConfig.network
    )
```

If you do this I think you'll get a good error that talks about a bad connection.